packages/appkit/src/client/appkit-base-client.ts

```diff
@@ -1001,6 +1003,7 @@ export abstract class AppKitBaseClient {
       if (network?.chainNamespace === ChainController.state.activeChain) {
         // If the network is unsupported and the user doesn't allow unsupported chains, we show the unsupported chain UI
         if (
+          OptionsController.state.showNetworkSwitcher &&
           !OptionsController.state.allowUnsupportedChain &&
           ChainController.state.activeCaipNetwork?.name === ConstantsUtil.UNSUPPORTED_NETWORK_NAME
         ) {
```

Comment:
Why do we show unsupported network only if network switcher enabled? Let's say user has Mainnet and Polygon enabled in AppKit. And via hooks or from wallet listeners they switched to OP, and if this prop is false, we're not going to show unsupported network UI right? Is this expected?

Reply:
If network switching is not allowed, we should still react to the network switch, but prompt the user to switch to the single network allowed (the connected one)